### PR TITLE
Add support for generalizedTime encoded certificate timestamps

### PR DESCRIPTION
**Pull request by mobileoverlord**
Support for generalized time encoded certificate validity was fixed in https://github.com/voltone/x509/commit/c1eb7bb6c280c27f4296809b3e8f1315bcdfb995

This updates the CA to support these changes.

### CODE REVIEW

File: test/test_helper.exs [hunk 1]
```diff
@@ -1,4 +1,8 @@
-File.rm_rf("test/tmp")
+tmp = Path.expand("test/tmp")
+
+File.rm_rf(tmp)
+File.mkdir_p(tmp)
```
Comment (fhunleth) [inline test/test_helper.exs]: This change doesn't look related to the generalized time. Did you want it?

Comment (mobileoverlord) [reply test/test_helper.exs]: I guess I could split this off into its own commit. It was a side effect of when I went to run the tests.
